### PR TITLE
added vaultKVUsePrefix on hashicorp vault secret

### PR DESCRIPTION
**Pull request by Banchio**
# Description

In the Hashicorp Vault secret store I added the vaultKVUsePrefix setting. if not passed maintain current behavior (assume value true and use the other setting to compute the actual prefix). If passed and is false then prefix is set to empty string so that this secret store can successfully use BulkGetSecret functionality.

## Issue reference

This PR should close the following issue: #925 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ X] Code compiles correctly
* [ X] Created/updated tests
* [X ] Extended the documentation: https://github.com/dapr/docs/pull/1543

### CODE REVIEW

**Comment by Banchio** [comment]
updated comment for documentation edit

**Comment by Banchio** [comment]
Is there a way to know if this can fit in v1.3 release (once discussed)? thanks a lot!

File: secretstores/hashicorp/vault/vault.go [hunk 1]
```diff
@@ -107,9 +108,14 @@ func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
 		return fmt.Errorf("token mount path and token both set")
 	}
 
+	vaultKVUsePrefix := props[componentVaultKVUsePrefix]
 	vaultKVPrefix := props[componentVaultKVPrefix]
-	if vaultKVPrefix == "" {
-		vaultKVPrefix = defaultVaultKVPrefix
+	if vaultKVUsePrefix == "false" {
+		vaultKVPrefix = ""
+	} else {
```
Comment (pkedy) [inline secretstores/hashicorp/vault/vault.go:115]: this could be a consolidated `else if`. Correct?

File: secretstores/hashicorp/vault/vault.go [hunk 2]
```diff
@@ -107,9 +108,14 @@ func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
 		return fmt.Errorf("token mount path and token both set")
 	}
 
+	vaultKVUsePrefix := props[componentVaultKVUsePrefix]
 	vaultKVPrefix := props[componentVaultKVPrefix]
-	if vaultKVPrefix == "" {
-		vaultKVPrefix = defaultVaultKVPrefix
+	if vaultKVUsePrefix == "false" {
```
Comment (pkedy) [inline secretstores/hashicorp/vault/vault.go:113]: You might consider using `strings.EqualFold(vaultKVUsePrefix, "false")` for case insensitivity.

**Review by pkedy — CHANGES_REQUESTED** [review]
2 small nitpicks

File: secretstores/hashicorp/vault/vault.go (hunk 2 above)
Comment (artursouza) [reply secretstores/hashicorp/vault/vault.go:113]: Why not use `strconv.ParseBool()`?

Comment (Banchio) [reply secretstores/hashicorp/vault/vault.go]: thanks, added a commit to consolidate if and use ParseBool to parse the setting.

**Review by pkedy — CHANGES_REQUESTED** [review]
Super small change to handle ParseBool error

File: secretstores/hashicorp/vault/vault.go [hunk 3]
```diff
@@ -107,8 +109,15 @@ func (v *vaultSecretStore) Init(metadata secretstores.Metadata) error {
 		return fmt.Errorf("token mount path and token both set")
 	}
 
+	vaultKVUsePrefix := props[componentVaultKVUsePrefix]
 	vaultKVPrefix := props[componentVaultKVPrefix]
-	if vaultKVPrefix == "" {
+	convertedVaultKVUsePrefix := true
+	if v, err := strconv.ParseBool(vaultKVUsePrefix); err == nil {
```
[the remark this answers is not in the document]
Comment (Banchio) [reply secretstores/hashicorp/vault/vault.go]: oh thanks so much, added string empty check and raise error if string not empty/true/false. I also added a test for that.

**Review by pkedy — APPROVED** [review]
LGTM and thanks for also creating the docs PR!

**Review by artursouza — APPROVED** [review]

**Pull request merged**